custom_components/solaredge_modbus/__init__.py

```diff
@@ -253,438 +253,435 @@ def read_modbus_data(self):
+        importvarhq2 = self.calculate_value(importvarhq2, energyvarsf)
+        importvarhq2a = self.calculate_value(importvarhq2a, energyvarsf)
+        importvarhq2b = self.calculate_value(importvarhq2b, energyvarsf)
+        importvarhq2c = self.calculate_value(importvarhq2c, energyvarsf)
+        importvarhq3 = self.calculate_value(importvarhq3, energyvarsf)
+        importvarhq3a = self.calculate_value(importvarhq3a, energyvarsf)
+        importvarhq3b = self.calculate_value(importvarhq3b, energyvarsf)
+        importvarhq3c = self.calculate_value(importvarhq3c, energyvarsf)
+        importvarhq4 = self.calculate_value(importvarhq4, energyvarsf)
+        importvarhq4a = self.calculate_value(importvarhq4a, energyvarsf)
+        importvarhq4b = self.calculate_value(importvarhq4b, energyvarsf)
+        importvarhq4c = self.calculate_value(importvarhq4c, energyvarsf)
+
+        self.data[meter_prefix + "importvarhq1"] = round(
+            importvarhq1, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq1a"] = round(
+            importvarhq1a, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq1b"] = round(
+            importvarhq1b, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq1c"] = round(
+            importvarhq1c, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq2"] = round(
+            importvarhq2, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq2a"] = round(
+            importvarhq2a, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq2b"] = round(
+            importvarhq2b, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq2c"] = round(
+            importvarhq2c, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq3"] = round(
+            importvarhq3, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq3a"] = round(
+            importvarhq3a, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq3b"] = round(
+            importvarhq3b, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq3c"] = round(
+            importvarhq3c, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq4"] = round(
+            importvarhq4, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq4a"] = round(
+            importvarhq4a, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq4b"] = round(
+            importvarhq4b, abs(energyvarsf)
+        )
+        self.data[meter_prefix + "importvarhq4c"] = round(
+            importvarhq4c, abs(energyvarsf)
+        )
+
+        return True
 
     def read_modbus_data_inverter(self):
         inverter_data = self.read_holding_registers(unit=self._address, address=40071, count=38)
         if not inverter_data.isError():
```

Comment:
think this 'not' need to be removed now according to switched if

Reply:
Good Catch. Must have crept back in when i was remerging from master